xarray/core/utils.py

```diff
@@ -542,69 +540,6 @@ def __repr__(self) -> str:
         return f"{type(self).__name__}({list(self)!r})"
 
 
-class NdimSizeLenMixin:
```

Comment:
Shall we just keep these in Xarray, and merge the mixin into the main NamedArray class? Seems like unnecessary indirection at the moment.

Reply:
I was wondering if those could just be rewritten without them.

Reply:
THe blast radius in Xarray seems decently big. 4 classes? Shall we keep the mixin in Xarray and avoid it in NamedArray?

Reply:
certainly, keeping the mixin in Xarray would greatly reduce the amount of code that needs to be rewritten in this PR. we can always address the removal or migration of the mixin in a separate PR later on.